predicators/nsrt_learning/strips_learning/base_strips_learner.py

```diff
@@ -32,6 +31,7 @@ def __init__(
         self._verbose = verbose
         self._num_segments = sum(len(t) for t in segmented_trajs)
         assert len(self._trajectories) == len(self._segmented_trajs)
+        self.timeout = timeout
```

Comment:
from our slack conversation: having a `timeout` here that is separate from `CFG.pnad_search_timeout` is a little awkward, but may be useful if we want to use a different timeout during predicate learning versus normal operator learning.

I am leaning toward recommending that we still just use `CFG.pnad_search_timeout` but I could be persuaded otherwise.

if we do keep the two separate, then I think we should make this private, and pass the timeout through to the creation of the search operators, rather than having the search operators access `self.learner.timeout`, because the latter obfuscates what is being used by the operators.